Commit basic config files

jbesoto opened the pull request:
## Changes:
- [FTPS-119](https://fiber-pts.atlassian.net/browse/FPTS-119)

## Considerations:
- Note that these files were created for the purpose of making them accessible to the team early in the development process.

jbesoto committed:
Commit basic config files

NxFerrara commented on config/pipe_config.py:
```diff
@@ -0,0 +1,3 @@
+# File paths for IPC pipes
+TAP_DATA_PIPE = '/tmp/FiberPTS/tapDataPipe'
```
I believe the /tmp folder is meant for non-persisting file data. For data or objects that need to persist, consider using directories like /var/run (for runtime data), /var/tmp (for temporarily stored files that should survive a reboot), or a directory in the user's home folder.

jbesoto replied:
Agreed. Changes made. Resolving this.

NxFerrara commented on config/pipe_config.py:
```diff
@@ -0,0 +1,3 @@
+# File paths for IPC pipes
+TAP_DATA_PIPE = '/tmp/FiberPTS/tapDataPipe'
+NFC_DATA_PIPE = '/tmp/FiberPTS/nfcDataPipe'
```
Same comment

NxFerrara commented on config/screen_config.py:
```diff
@@ -0,0 +1,15 @@
+# FONT_COLOR =
+# DASHBOARD_BG_COLOR = 
+
+# Pop-up message configurations
```
Add POPUP_FONT_FAMILY, and change line 8 and 9 to be consistently named.

jbesoto replied:
I just figured that we would use the same font. I will add it regardless as it provides flexibility.

NxFerrara commented on config/screen_config.py:
```diff
@@ -0,0 +1,15 @@
+# Dashboard configurations
```
Make sure all configurations have DASHBOARD at the beginning to differentiate from the Popup configs.

jbesoto replied:
Agreed. Changes made

NxFerrara commented on config/screen_config.py:
```diff
@@ -0,0 +1,15 @@
+# GOOD_TAP_BG_COLOR =
+# BAD_TAP_BG_COLOR =
+
+# SET_ORDER_BG_COLOR =
```
Maybe do/add the following for consistency:

GOOD_ORDER_BG_COLOR
GOOD_EMPLOYEE_BG_COLOR
BAD_NFC_BG_COLOR

jbesoto replied:
- Prepended all general pop-up configurations with `POPUP_` for consistency. 
- Added `POPUP_ERROR_BG_COLOR` and `POPUP_WARNING_BG_COLOR` to represent errors (e.g. NFC read failed) and warnings (e.g. tap within debounce time)
- Since there is already representation for error and warning, only good NFC and touch sensor taps configurations were created. Here:
```py
# Tap event pop-up configurations
TAP_EVENT_BG_COLOR = ''

# NFC tap pop-up configurations
ORDER_SET_BG_COLOR = ''
EMPLOYEE_SET_BG_COLOR = ''
```

NxFerrara replied:
Alternative:
GOOD_ORDER_NFC_BG_COLOR
GOOD_EMPLOYEE_NFC_BG_COLOR

NxFerrara reviewed (changes requested):
I like this setup for global system variable management! 

I have a few considerations, so let me know what you think.

Also, I presume that the config variables that are commented are tbd.

jbesoto committed:
Change file paths to pipes

/opt was selected as it provides isolation from standard software directories (e.g. /usr), allows for simpler permission and access management, and it is conventionally used for storing third-party software (like ours).

jbesoto committed:
Fix config variable naming

NxFerrara commented on config/pipe_config.py:
```diff
@@ -0,0 +1,3 @@
+# File paths for IPC pipes
```
I was just thinking. Maybe we have a PIPE_PATH variable for where we want to store the pipes like `'/FiberPTS/pipes'` and a list of pipe file names assigned to PIPE_NAMES such as `['tapDataPipe', 'nfcDataPipe']`.

UPDATE: Okay I realized that you may not do PIPE_NAMES as a list and instead just list out variables like you've been doing since you need to get this information from some of the programs, however I think the PIPE_PATH variable is still useful so that we can just change all the pipes to a particular path at once, keeping it modular.

NxFerrara replied:
Also maybe change the pipe names since they do specific things like `'tapDataPipe'` could be renamed to `'touchSensorToScreenPipe'` since the pipe describes an inter-process communication bridge between two specific processes. I know it doesn't look pretty, but it feels more functionally correct.

NxFerrara replied:
I would not assume the `'/opt'` absolute path, just because 1. we don't need to and 2. it makes choosing a different directory for whatever reason a bit harder

jbesoto replied:
The reason why I chose this name, say `tap_data_pipe`, was to convey that tap data is being transmitted through that pipe. Any module can read from this pipe, so a general name to describe the information being transmitted made sense.

NxFerrara replied:
The reason is that pipes should only be a communication with a single process, so, for instance, if the screen is reading from `'tap_data_pipe'`, then no other process should read from that pipe ever to avoid concurrency issues since every read removes the data from the pipe (like dequeuing), so information may be lost to one of the concurrently reading processes.

jbesoto replied:
> I would not assume the `'/opt'` absolute path, just because 1. we don't need to and 2. it makes choosing a different directory for whatever reason a bit harder

Can you elaborate more on this? Not sure I understand.

jbesoto replied:
> I was just thinking. Maybe we have a PIPE_PATH variable for where we want to store the pipes like `'/FiberPTS/pipes'` and a list of pipe file names assigned to PIPE_NAMES such as `['tapDataPipe', 'nfcDataPipe']`.

Makes sense. I will add it.

NxFerrara replied:
All that is needed for the pipe path is where in `'/FiberPTS'` the pipes will reside since we agreed that during setup, the pipes are just created within the FiberPTS application folder. No need to include where FiberPTS is, since that is just unnecessary information. If needed, can't you just use a relative pathing like `'../FiberPTS/pipes'`?

jbesoto replied:
Before committing, here is what I understood from your comments:
```py
# File paths for IPC pipes
PIPE_PATH = '/opt/FiberPTS/pipes'
TAP_DATA_PIPE = f'{PIPE_PATH}/tap_data_pipe'
NFC_DATA_PIPE = f'{PIPE_PATH}/nfc_data_pipe'
```
Let me know your thoughts.

NxFerrara reviewed (commented):
Just a comment (forget this)

jbesoto committed:
Fix pipe naming

jbesoto committed:
Store pipe paths in a shared env file

- Pipe paths were stored in a shared env file
- Dedicated utility script was created to loading envvars
This allows for easy access of named pipe file paths during IoT setup and within Python scripts.

NxFerrara commented on .env.shared:
```diff
@@ -0,0 +1,4 @@
+# File paths to named pipes
+FOLDER_PATH=/opt/FiberPTS/pipes
```
Last change: change to `PIPE_PATH` or `PIPE_FOLDER_PATH` for clarity.

Otherwise, amazing!

jbesoto replied:
Agreed.

NxFerrara commented on config/screen_config.py:
Great adjustments. I like your thinking.

One thing is that we need to discuss how we store this config information. For instance, should we store it in the `.env.shared` or a new one called `.env.config` and, just like you did with `pipe_paths.py`, adjust `screen_config.py` so that it pulls that config info.

This should be considered for all `.py` config files.

jbesoto replied:
Why would we want to do this? The only reason why we created the `.env.shared` is because both the setup script and other Python modules will use this. If only Python modules uses configurations, it does not make sense to store it in a `.env` file. 

That said, I disagree that we should consider that for all `.py` config files. Recall that `.env` files are not used to store store module-specific information but rather to keep sensitive or environment-specific variables, like API keys, database passwords, and endpoints, separate from the codebase.

NxFerrara replied:
Okay, so all setup configurations that the setup script(s) would need would be in the `.env.shared` file? If so, I totally agree with you.

NxFerrara commented on src/utils/pipe_paths.py:
Nice, this is a utility. Makes sense. I like the use of environment file to store config information.

NxFerrara reviewed (changes requested):
Overall, great changes. I have some considerations on how we are going to store the config information along with minor naming changes.

jbesoto committed:
Fix pipe folder naming for clarity

NxFerrara reviewed (approved):
Perfect. Nice work.
